bin/node-template/pallets/template/src/lib.rs

```diff
@@ -57,7 +57,14 @@ pub mod pallet {
 	}
 
 	#[pallet::hooks]
-	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_finalize(block_number: T::BlockNumber) {
+			debug::info!("on finalize");
+		}
+		fn on_idle(block_number: T::BlockNumber) {
```

Comment:
i would add an extra parameter to describe how much weight is left for use in the block.